### PR TITLE
Selenium: Rework the 'CheckoutToRemoteBranchTest' to run in multi-threaded mode

### PR DESCRIPTION
**Pull request by artaleks9**
### What does this PR do?
* Improve and rework selenium test 'CheckoutToRemoteBranchTest'  
* Use the _kohsuke Github_ library
* Add the method 'addRefFromMaster()' to the 'TestGitHubRepository'
* Incapsulate the methods 'deleteFileOnGitHub()' and 'changeContentOnGitHub()' in the 'TestGitHubRepository' 

### What issues does this PR fix or reference?
#8986, #8153

### CODE REVIEW

**Comment by artaleks9** [comment]
ci-test

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 1]
```diff
@@ -85,6 +86,17 @@ public String getName() {
     return repoName;
   }
 
+  /**
+   * Create reference to branch, tag, ... from master branch.
+   *
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: Create >>>>  Creates

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 2]
```diff
@@ -85,6 +86,17 @@ public String getName() {
...
+   * @param refName
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: use please full name and complete description

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java [hunk 3]
```diff
@@ -10,273 +10,143 @@
...
   }
 
-  private void checkChangesIntoCompareForm(String expText) {
-    git.clickOnGroupCompareButton();
-    git.waitGitCompareFormIsOpen();
-    git.waitExpTextIntoCompareLeftEditor(expText);
-    git.waitTextNotPresentIntoCompareRightEditor(expText);
+  private void changeContentOnGithubSide(String pathToContent, String content, String branchName)
+      throws IOException {
+    testRepo
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: It is better to move this method to separate method `update(Path relativePathToFile, byte[] newContent, String branchName)` in _TestGitHubRepository_ class.
Commit message ought to be simple and not to hold new content itself.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java [hunk 4]
```diff
@@ -10,273 +10,143 @@
...
-    git.waitGitStatusBarWithMess(pushedMessage);
+    performGitPull();
 
-    events.clickEventLogBtn();
-    events.waitExpectedMessage(PUSH_MSG);
+    // change content in the test repo on GitHub
+    changeContentOnGithubSide(pathJavaFile, changeContent, SECOND_BRANCH);
+    changeContentOnGithubSide(pathJspFile, changeContent, SECOND_BRANCH);
 
-    // import from hosted git repository the second project
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: Could you, please, explain why did you remove next verifications:
1. Check presence of previous changes
1. Check the change in git compare form

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java [hunk 5]
```diff
@@ -10,273 +10,143 @@
...
-        "Successfully pushed to git@github.com:"
-            + gitHubUsername
-            + "/springProjectWithSeveralBranches.git";
-    git.waitGitStatusBarWithMess(pushedMessage);
+    performGitPull();
 
-    events.clickEventLogBtn();
-    events.waitExpectedMessage(PUSH_MSG);
+    // change content in the test repo on GitHub
+    changeContentOnGithubSide(pathJavaFile, changeContent, SECOND_BRANCH);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: Why did you replace manual modification of files by changing them through the `TestGitHubRepository`?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java (hunk 4 above)
Comment (artaleks9) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: 1. Check presence of previous changes is performed (see line 115)
1. Check the change in the git compare form is deleted, because it is not a main goal of this selenium test, and it is an unneeded, on my opinion. Also there is the selenium test `GitcompareForm` where is performed needed checks

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java (hunk 5 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: I see at least one method here which checks only here: `git.selectPushRemoteBranchName(SECOND_BRANCH);`.

Comment (artaleks9) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/CheckoutToRemoteBranchTest.java]: I have reverted the check git history and the performing git push to second remote branch from UI

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 6]
```diff
@@ -115,6 +127,21 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
...
 
+  /**
+   * Change content of the file
+   *
+   * @param pathToFile path to specified file
+   * @param content content to change
+   * @throws IOException
+   */
+  public void changeFileContent(String pathToFile, String content) throws IOException {
+    ghRepo.getFileContent(String.format("/%s", pathToFile)).update(content, "change - " + content);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: Adding content to the commit message is not a good idea. Also, it's better to have unified commit message like: `Change file <pathToFile>`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 7]
```diff
@@ -115,6 +127,21 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
...
+  }
+
+  public void deleteFile(String pathToContent) throws IOException {
+    ghRepo.getFileContent(pathToContent).delete("delete-file");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: it's better to have unified commit message like: `Delete file <pathToFile>`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 8]
```diff
@@ -115,6 +127,21 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
...
+   *
+   * @param pathToFile path to specified file
+   * @param content content to change
+   * @throws IOException
+   */
+  public void changeFileContent(String pathToFile, String content) throws IOException {
+    ghRepo.getFileContent(String.format("/%s", pathToFile)).update(content, "change - " + content);
+  }
+
+  public void deleteFile(String pathToContent) throws IOException {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: It's about _pathToFile_, not _pathToContent_.

Comment (artaleks9) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: Correct according to the last review

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 9]
```diff
@@ -135,11 +135,26 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
    * @throws IOException
    */
   public void changeFileContent(String pathToFile, String content) throws IOException {
-    ghRepo.getFileContent(String.format("/%s", pathToFile)).update(content, "change - " + content);
+    ghRepo
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: It could be simplified as follow: `changeFileContent(pathToFile, content, format("Change file %s", pathToFile));`

It also removes code duplication.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/FetchAndMergeRemoteBranchIntoLocalTest.java [hunk 10]
```diff
@@ -86,8 +86,10 @@ public void fetchUpdatesAndMergeRemoteBranchTolocal() throws IOException {
     // change content in the test repo on GitHub
     testRepo.deleteFile(String.format("%s/%s", pathToJspFile, jspFile));
     testRepo.changeFileContent(
-        String.format("%s/%s.java", pathToJavaFile, javaFile), CHANGE_CONTENT);
-    testRepo.changeFileContent(textFile, CHANGE_CONTENT);
+        String.format("%s/%s.java", pathToJavaFile, javaFile),
+        CHANGE_CONTENT,
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/FetchAndMergeRemoteBranchIntoLocalTest.java]: `CHANGE_CONTENT` > `NEW_CONTENT`

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 11]
```diff
@@ -135,11 +135,26 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
...
+        .getFileContent(String.format("/%s", pathToFile))
+        .update(content, "change - " + pathToFile);
   }
 
-  public void deleteFile(String pathToContent) throws IOException {
-    ghRepo.getFileContent(pathToContent).delete("delete-file");
+  /**
+   * Change content of the file
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: `Change` > `Changes`

**Comment by artaleks9** [comment]
Correct according to the last review

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java [hunk 12]
```diff
@@ -135,9 +135,7 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
    * @throws IOException
    */
   public void changeFileContent(String pathToFile, String content) throws IOException {
-    ghRepo
-        .getFileContent(String.format("/%s", pathToFile))
-        .update(content, "change - " + pathToFile);
+    changeFileContent(pathToFile, content, format("Change file-%s", pathToFile));
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java]: it seems there is redundant "-" in commit message which makes commit message misleading in part of file path.

**Review by dmytro-ndp — APPROVED** [review]
